### PR TITLE
Add the DPI for the f-divergence in the absolutely continuous case.

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
- Add `Measure.add_sub_cancel`.
- Add `measure_sub_singularPart` and `measure_sub_rnDeriv`.
- Add `absolutelyContinuous_comp`, `absolutelyContinuous_comp_left`, `absolutelyContinuous_comp_right`.
- Add `statInfoFun_neg_neg`, `derivAtTop_statInfoFun_eq`, `derivAtTop_statInfoFun_nonneg`.
- Rename `stronglymeasurable_statInfoFun` to `stronglyMeasurable_statInfoFun`.
- Add `boolMeasure_apply_univ`and an instance for `IsFiniteMeasure` of `boolMeasure`.
- Add `statInfo_ne_top`. 
- Add `toReal_statInfo_eq_integral_abs`, `statInfo_eq_min_sub_iInf_measurableSet`.
- Add `fDiv_statInfoFun_nonneg`, `fDiv_statInfoFun_stronglyMeasurable`, `fDiv_statInfoFun_eq_zero_of_nonneg_of_nonpos`, `fDiv_statInfoFun_eq_zero_of_nonpos_of_pos`.
- Add `fDiv_statInfoFun_eq_StatInfo_of_nonneg_of_le`, `fDiv_statInfoFun_eq_StatInfo_of_nonneg_of_gt`, `fDiv_statInfoFun_eq_StatInfo_of_nonneg`, `fDiv_statInfoFun_ne_top_of_nonneg`. 
- Generalize `integral_statInfoFun_curvatureMeasure` and add `integral_statInfoFun_curvatureMeasure'`.
- Add `fDiv_eq_fDiv_centeredFunction`, `lintegral_f_rnDeriv_eq_lintegralfDiv_statInfoFun_of_absolutelyContinuous`, `fDiv_ne_top_iff_integrable_fDiv_statInfoFun_of_absolutelyContinuous'`, `fDiv_ne_top_iff_integrable_fDiv_statInfoFun_of_absolutelyContinuous`, `integrable_f_rnDeriv_iff_integrable_fDiv_statInfoFun_of_absolutelyContinuous`.
- Generalize `fDiv_eq_integral_fDiv_statInfoFun_of_absolutelyContinuous` and add `fDiv_eq_integral_fDiv_statInfoFun_of_absolutelyContinuous'`.
- Add `fDiv_eq_lintegral_fDiv_statInfoFun_of_absolutelyContinuous`.
- Add the DPI: `fDiv_statInfoFun_comp_right_le`, `fDiv_statInfoFun_comp_right_le'`, `fDiv_comp_right_le_of_absolutelyContinuous`.
- Cleanup.

Depends on #117 and #118.

### CODE REVIEW

File: TestingLowerBounds/Divergences/StatInfo.lean [hunk 1]
```diff
@@ -147,7 +151,9 @@ lemma toReal_statInfo_eq_min_sub_integral (μ ν : Measure 𝒳) [IsFiniteMeasur
   rw [toReal_bayesBinaryRisk_eq_integral_min,
     MonotoneOn.map_min (fun _ _ _ hb hab ↦ ENNReal.toReal_mono hb hab) hμ hν]
 
-lemma toReal_statInfo_eq_min_sub_integral' {μ ν ζ : Measure 𝒳} [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+#check Measure.rnDeriv_eq_div'
+
```
Comment (RemyDegenne) [inline TestingLowerBounds/Divergences/StatInfo.lean]:
```suggestion
```
Please scroll through the changes of your PRs before opening them. You often leave things like this or stray examples or comments.

File: TestingLowerBounds/Divergences/StatInfo.lean [hunk 2]
```diff
@@ -537,39 +652,196 @@ lemma fDiv_statInfoFun_eq_integral_abs_of_nonpos_of_gt [IsFiniteMeasure μ] [IsF
...
+  rcases le_or_lt γ β with (hβγ | hβγ)
+  · rw [fDiv_statInfoFun_eq_StatInfo_of_nonneg_of_le hβ hγ hβγ, if_pos hβγ, neg_one_mul,
+      EReal.neg_sub, add_comm (-_), sub_eq_add_neg, add_assoc]
+    · exact Or.inl <| EReal.add_top_iff_ne_bot.mp rfl
+    · exact Or.inl <| Ne.symm (ne_of_beq_false rfl)
+  · rw [fDiv_statInfoFun_eq_StatInfo_of_nonneg_of_gt hβ hγ hβγ, if_neg hβγ.not_le, one_mul,
+      add_sub_assoc]
+
+lemma fDiv_statInfoFun_ne_top_of_nonneg [IsFiniteMeasure μ] [IsFiniteMeasure ν] :
+    fDiv (statInfoFun β γ) μ ν ≠ ⊤ := by
```
Comment (RemyDegenne) [inline TestingLowerBounds/Divergences/StatInfo.lean]:
```suggestion
lemma fDiv_statInfoFun_ne_top [IsFiniteMeasure μ] [IsFiniteMeasure ν] :
    fDiv (statInfoFun β γ) μ ν ≠ ⊤ := by
```
Why `of_nonneg`?

File: TestingLowerBounds/Divergences/StatInfo.lean [hunk 3]
```diff
@@ -537,39 +652,196 @@ lemma fDiv_statInfoFun_eq_integral_abs_of_nonpos_of_gt [IsFiniteMeasure μ] [IsF
...
--- TODO: think about the case when the function is not integrable (`h_int`).
--- Can we prove that in this case the rhs is also not integrable?
-lemma fDiv_eq_integral_fDiv_statInfoFun_of_absolutelyContinuous
+lemma integral_statInfoFun_curvatureMeasure' (hf_cvx : ConvexOn ℝ univ f) (hf_cont : Continuous f)
+    (hf_one : f 1 = 0) (hfderiv_one : rightDeriv f 1 = 0) :
+    ∫ y, statInfoFun 1 y t ∂(curvatureMeasure f) = f t := by
+  rw [integral_statInfoFun_curvatureMeasure hf_cvx hf_cont, hf_one, hfderiv_one, sub_zero, zero_mul,
+    sub_zero]
+
+lemma fDiv_eq_fDiv_centeredFunction [IsFiniteMeasure μ] [IsFiniteMeasure ν]
```
Comment (RemyDegenne) [inline TestingLowerBounds/Divergences/StatInfo.lean]: That should be in FDiv.Basic

**Review by RemyDegenne — APPROVED** [review]
Please resolve the comments then merge.
This looks great!

File: TestingLowerBounds/Divergences/StatInfo.lean (hunk 2 above)
Comment (LorenzoLuccioli) [reply TestingLowerBounds/Divergences/StatInfo.lean]: I think that I first wrote another proof of this, involving `fDiv_statInfoFun_eq_StatInfo_of_nonneg`, that needed the hypotheses `(hβ : 0 ≤ β) (hγ : 0 ≤ γ)`, then I realized that there was a much shorter proof without those hypotheses, but I forgot to change the name accordingly.

**Pull request merged**